src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -0,0 +1,1850 @@
+    TLV::TLVWriter writer;
+
+    writer.Init(buffer);
+    ReturnErrorOnFailure(viewport.Encode(writer, TLV::AnonymousTag()));
+
+    auto path = ConcreteAttributePath(mEndpointId, CameraAvStreamManagement::Id, Attributes::Viewport::Id);
+    return GetSafeAttributePersistenceProvider()->SafeWriteValue(path, ByteSpan(buffer));
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::LoadViewport(ViewportStruct & viewport)
+{
+    uint8_t buffer[kViewportStructMaxSerializedSize];
+    MutableByteSpan bufferSpan(buffer);
+    uint16_t size = static_cast<uint16_t>(bufferSpan.size());
+
+    auto path = ConcreteAttributePath(mEndpointId, CameraAvStreamManagement::Id, Attributes::Viewport::Id);
+    ReturnErrorOnFailure(GetSafeAttributePersistenceProvider()->SafeReadValue(path, bufferSpan));
+    bufferSpan.reduce_size(size);
+
+    TLV::TLVReader reader;
+
+    reader.Init(bufferSpan);
+    ReturnErrorOnFailure(reader.Next(TLV::AnonymousTag()));
+    ReturnErrorOnFailure(viewport.Decode(reader));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::StoreRankedVideoStreamPriorities()
+{
+    uint8_t buffer[kRankedVideoStreamPrioritiesTlvSize];
+    MutableByteSpan bufferSpan(buffer);
+    TLV::TLVWriter writer;
+    writer.Init(bufferSpan);
+
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Array, arrayType));
+
+    for (size_t i = 0; i < mRankedVideoStreamPriorities.size(); i++)
+    {
+        ReturnErrorOnFailure(writer.Put(TLV::AnonymousTag(), mRankedVideoStreamPriorities[i]));
+    }
+    ReturnErrorOnFailure(writer.EndContainer(arrayType));
+
+    bufferSpan.reduce_size(writer.GetLengthWritten());
+
+    auto path = ConcreteAttributePath(mEndpointId, CameraAvStreamManagement::Id, Attributes::RankedVideoStreamPrioritiesList::Id);
+    return GetSafeAttributePersistenceProvider()->SafeWriteValue(path, bufferSpan);
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::LoadRankedVideoStreamPriorities()
+{
+    uint8_t buffer[kRankedVideoStreamPrioritiesTlvSize];
+    MutableByteSpan bufferSpan(buffer);
+    uint16_t size = static_cast<uint16_t>(bufferSpan.size());
+
+    auto path = ConcreteAttributePath(mEndpointId, CameraAvStreamManagement::Id, Attributes::RankedVideoStreamPrioritiesList::Id);
+    ReturnErrorOnFailure(GetSafeAttributePersistenceProvider()->SafeReadValue(path, bufferSpan));
+    bufferSpan.reduce_size(size);
+
+    TLV::TLVReader reader;
+    reader.Init(bufferSpan);
+
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::AnonymousTag()));
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(reader.EnterContainer(arrayType));
+
+    uint8_t i = 0;
+    CHIP_ERROR err;
+    while ((err = reader.Next(TLV::kTLVType_UnsignedInteger, TLV::AnonymousTag())) == CHIP_NO_ERROR && i < kNumOfStreamUsageTypes)
+    {
+        ReturnErrorOnFailure(reader.Get(mRankedVideoStreamPriorities[i]));
+        i++;
+    }
+
+    VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
```

Comment:
OK, so here if the stored TLV had more than kNumOfStreamUsageTypes entries, we will return CHIP_NO_ERROR at this point, right, without doing the ExitContainer/VerifyEndOfContainer below.  Is that the expected behavior?

Reply:
kNumOfStreamUsageTypes is expected to hold the max number of supported stream usages. So, the stored list is not expected to have more than kNumOfStreamUsageTypes entries. However, we have added a new attribute `SupportedStreamUsages` and the `RankedStreamPriorities` list should conform to that. I have filed an issue(https://github.com/project-chip/connectedhomeip/issues/37933) for adding that logic.